pulser/register.py

```diff
@@ -133,7 +150,9 @@ def _draw_2D(
         if with_labels:
             # Determine which labels would overlap and merge those
             plot_pos = list(pos[:, (ix, iy)])
-            plot_ids = [f"{i}" for i in ids]
+            plot_ids = [
+                f"[{i}]" if i in masked_qubits else f"{i}" for i in ids
+            ]
```

Comment:
What if instead you join all masked qubits under the same brackets, irrespectively of the order? E.g; if '0', '1' and '4' are masked the label would be: '2, 3, [0, 1, 4]'

Then you can just group all the masked qubits together.

Reply:
I guess the problem here is that at that line you still don't know if their label will be merged, because that only happens if they are drawn very close together (like in the case of 2d projections of 3d registers). 
For example if you look at the picture I posted in the previous comment, `[1,2]` are merged in the x-y view but they are not in the x-z and y-z.

Reply:
Well yeah, you wouldn't be doing it here exactly. When you're deciding which qubits go in a label, you also check whether they are masked. By the end you have a group of masked and unmasked and you can compose your label. Doesn't this work?